sdks/python/apache_beam/dataframe/convert.py

```diff
@@ -0,0 +1,71 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import inspect
+
+from apache_beam import pvalue
+from apache_beam.dataframe import expressions
+from apache_beam.dataframe import frame_base
+from apache_beam.dataframe import transforms
+
+
+def to_dataframe(pc):
+  pass
+
+
+# TODO: Or should this be called as_dataframe?
```

Comment:
I like `as_dataframe` if it were a method on `PCollection` since it's fluent - `df = pcol.as_dataframe()`

Similarly, below `from_dataframe` would be fluent as a static method on `PCollection`, but `PCollection.from_dataframe` feels too verbose.

Would you agree we want something like the following to be as easy and intuitive as possible?
```py
pcol = p | "Read from Source" >> beam.io.SomeSchemaSource(foo)

df = pcol.as_dataframe()
df_agg = df[df["measurement" > threshold]].groupby("id").count()

PCollection.from_dataframe(df_agg) | "Write to Sink" >> beam.io.SomeSink(bar)
```

Reply:
So far we haven't added any methods to PCollection, but I'm open to the idea (thought it'd be a big change to the API that should be done wholistically, see https://lists.apache.org/thread.html/fcb422d61437a634662b24100d4e2d46a940ee766848b699023081d9%40%3Cdev.beam.apache.org%3E ) For now, at least, it seems a bit much to make dataframe-methods on PCollection itself. 

Short of that, can you think of any fluent styles for

```
from apache_beam import dataframe as ???
...
pcol = p | "Read from Source" >> beam.io.SomeSchemaSource(foo)
df = ???
```

Reply:
Yeah definitely fair to not make changes to PCollection yet. Another option would be methods on DeferredDataframe, like #as_pcollection():

```python
pcol = p | "Read from Source" >> beam.io.SomeSchemaSource(foo)

df = to_dataframe(pcol)
df_agg = df[df["measurement" > threshold]].groupby("id").count()

df_agg.as_pcollection() | "Write to Sink" >> beam.io.SomeSink(bar)
```

This is just a bikeshed though. This is still very experimental so I don't think we're locking ourselves into anything.

Reply:
Yeah, it's all experimental at this point. We can keep churning on the best API while we flesh out functionality.